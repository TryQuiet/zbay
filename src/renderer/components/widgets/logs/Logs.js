import React, { useState } from 'react'
import * as R from 'ramda'
import classnames from 'classnames'
// import { AutoSizer } from 'react-virtualized'
import { Scrollbars } from 'react-custom-scrollbars'
import { shell } from 'electron'
import PropTypes from 'prop-types'

import { withStyles } from '@material-ui/core/styles'

import Grid from '@material-ui/core/Grid'
import { Typography } from '@material-ui/core'

import IconButton from '../../ui/IconButton'
import Icon from '../../ui/Icon'
import ExitIcon from '../../../static/images/logs/exit.svg'

const styles = theme => ({
  root: {
    width: '315px',
    height: '100vh',
    backgroundColor: theme.palette.colors.logsActiveDark
  },
  title: {
    lineHeight: '18px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: theme.palette.colors.logsTitleGray,
    marginTop: 20
  },
  topBar: {
    width: '100%',
    height: 58,
    paddingLeft: 20,
    paddingRight: 25,
    backgroundColor: theme.palette.colors.logsDark
  },
  iconButton: {
    marginTop: 20,
    padding: 0
  },
  tabBar: {
    color: theme.palette.colors.logsTitleGray,
    backgroundColor: theme.palette.colors.logsDark
  },
  tab: {
    marginRight: 1,
    cursor: 'pointer',
    padding: '4px 16px',
    backgroundColor: theme.palette.colors.logsInactiveDark
  },
  tabText: {
    lineHeight: '18px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: theme.palette.colors.logsTabWhite
  },
  activeTab: {
    backgroundColor: theme.palette.colors.logsActiveDark
  },
  mainLogsWindow: {
    margin: '0px 0px',
    width: '315px',
    height: '100%',
    backgroundColor: theme.palette.colors.logsActiveDark
  },
  logsItem: {
    width: '100%',
    padding: '0px 8px'
  },
  logsLine: {
    fontFamily: 'Menlo Regular',
    lineHeight: '18px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: theme.palette.colors.logsTitleGray,
    wordWrap: 'break-word'
  },
  transactionLine: {
    fontFamily: 'Menlo Regular',
    lineHeight: '18px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: theme.palette.colors.logsTitleGray,
    wordWrap: 'break-word',
    cursor: 'pointer',
    '&:hover': {
      color: theme.palette.colors.lushSky
    }
  },
  verticalScrollBar: {
    position: 'absolute',
    top: 0,
    right: 0,
    height: '100%',
    display: 'flex',
    'justify-content': 'center',
    backgroundColor: theme.palette.colors.logsScrollBar
  },
  renderThumbVertical: {
    borderRadius: 25,
    backgroundColor: theme.palette.colors.logsScrollBarThumb,
    cursor: 'pointer'
  },
  innerContent: {
    marginTop: 8,
    paddingRight: 15,
    paddingBottom: 5
  }
})

const LogsTypes = {
  TRANSACTIONS: 'TRANSACTIONS',
  APPLICATION_LOGS: 'APPLICATION_LOGS',
  NODE_DEBUG: 'NODE_DEBUG'
}

export const LogsComponent = ({ classes, debugLogs, closeLogsWindow, applicationLogs, transactionsLogs, height }) => {
  const logs = {
    [LogsTypes.TRANSACTIONS]: transactionsLogs,
    [LogsTypes.NODE_DEBUG]: debugLogs,
    [LogsTypes.APPLICATION_LOGS]: applicationLogs
  }
  const messagesEndRef = React.useRef(null)
  const [currentActiveTab, setActiveTab] = useState(LogsTypes.NODE_DEBUG)
  const changeTab = (tab) => {
    setActiveTab(tab)
    messagesEndRef.current.scrollToBottom()
  }
  React.useEffect(() => {
    if (messagesEndRef.current) {
      messagesEndRef.current.scrollToBottom()
    }
  })
  return (
    <Grid container className={classes.root} alignContent={'flex-start'} item>
      <Grid container className={classes.topBar} justify={'space-between'} item>
        <Grid item>
          <Typography variant={'subtitle1'} className={classes.title}>Logs</Typography>
        </Grid>
        <Grid item>
          <IconButton className={classes.iconButton} onClick={closeLogsWindow}>
            <Icon src={ExitIcon} />
          </IconButton>
        </Grid>
      </Grid>
      <Grid container direction={'row'} className={classes.tabBar} justify={'flex-start'} wrap={'nowrap'} item>
        <Grid item onClick={() => changeTab(LogsTypes.TRANSACTIONS)} className={classnames(classes.tab, {
          [classes.activeTab]: currentActiveTab === LogsTypes.TRANSACTIONS
        })}>
          <Typography variant={'caption'} className={classes.tabText}>Transactions</Typography>
        </Grid>
        <Grid item onClick={() => changeTab(LogsTypes.NODE_DEBUG)} className={classnames(classes.tab, {
          [classes.activeTab]: currentActiveTab === LogsTypes.NODE_DEBUG
        })}>
          <Typography variant={'caption'} className={classes.tabText}>Zcashd</Typography>
        </Grid>
        <Grid item onClick={() => changeTab(LogsTypes.APPLICATION_LOGS)} className={classnames(classes.tab, {
          [classes.activeTab]: currentActiveTab === LogsTypes.APPLICATION_LOGS
        })}>
          <Typography variant={'caption'} className={classes.tabText}>Application</Typography>
        </Grid>
      </Grid>
      <Grid container className={classes.mainLogsWindow} item>
        <Scrollbars
          ref={messagesEndRef}
          autoHideTimeout={500}
          style={{ width: 315, height: height - 90 }}
          renderTrackVertical={props => <div {...props} style={{ width: '14px' }} className={classes.verticalScrollBar} />}
          renderThumbVertical={props => <div {...props} style={{ width: '8px' }} className={classes.renderThumbVertical} />}
        >
          <div className={classes.innerContent}>
            {logs[currentActiveTab].map((logLine, i) => currentActiveTab === LogsTypes.TRANSACTIONS
              ? <Grid item key={i} className={classes.logsItem}><Typography onClick={() => shell.openExternal(`https://explorer.zcha.in/transactions/${logLine}`)} className={classes.transactionLine} variant={'caption'}>{logLine}</Typography></Grid>
              : <Grid item key={i} className={classes.logsItem}><Typography className={classes.logsLine} variant={'caption'} key={i}>{logLine}</Typography></Grid>)
            }
          </div>
        </Scrollbars>
          )}
      </Grid>
    </Grid>
  )
}

LogsComponent.propTypes = {
  classes: PropTypes.object.isRequired,
  debugLogs: PropTypes.array.isRequired,
  applicationLogs: PropTypes.array.isRequired,
  transactionsLogs: PropTypes.array.isRequired,
  closeLogsWindow: PropTypes.func.isRequired,
  height: PropTypes.number
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(LogsComponent)
