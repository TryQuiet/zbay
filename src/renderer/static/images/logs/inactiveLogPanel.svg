<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g opacity="0.5">
<path d="M8.32544 9.5C8.0493 9.5 7.82544 9.72386 7.82544 10C7.82544 10.2761 8.0493 10.5 8.32544 10.5L8.32544 9.5ZM11.5 10.5C11.7761 10.5 12 10.2761 12 10C12 9.72386 11.7761 9.5 11.5 9.5L11.5 10.5ZM8.32544 10.5L11.5 10.5L11.5 9.5L8.32544 9.5L8.32544 10.5Z" fill="white"/>
<path d="M4.43604 5.48047L6.6958 7.74023L4.43604 10" stroke="white" stroke-linecap="round"/>
<rect x="0.5" y="0.5" width="15" height="15" rx="3.5" stroke="white"/>
</g>
</svg>
