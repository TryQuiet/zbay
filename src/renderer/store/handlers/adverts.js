import BigNumber from 'bignumber.js'

import identitySelectors from '../selectors/identity'
import channelSelectors from '../selectors/channel'
import directMessageChannelSelectors from '../selectors/directMessageChannel'
import offersHandlers from '../../store/handlers/offers'
import { messages } from '../../zbay'
import { getClient } from '../../zcash'
import notificationsHandlers from './notifications'
import directMessagesQueueHandlers, {
  checkConfirmationNumber
} from './directMessagesQueue'
import { errorNotification, successNotification } from './utils'
import operationsHandlers, { operationTypes } from './operations'
import contactsHandlers from './contacts'
import { messageType } from '../../../shared/static'

const handleSend = ({ values }) => async (dispatch, getState) => {
  const data = {
    tag: values.tag,
    background: values.background.toString(),
    title: values.title,
    provideShipping: values.shippingInfo ? '1' : '0',
    amount: values.usd,
    description: values.description
  }
  const identityAddress = identitySelectors.address(getState())
  const channel =
    channelSelectors.data(getState()) &&
    channelSelectors.data(getState()).toJS()
  const directChannelAddress = directMessageChannelSelectors.targetRecipientAddress(
    getState()
  )
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.AD,
      data: data,
      spent: channel ? channel.advertFee.toString() : '0'
    },
    privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: channel ? channel.address : directChannelAddress,
    identityAddress
  })
  try {
    const opId = await getClient().payment.send(transfer)
    await dispatch(
      operationsHandlers.epics.observeOperation({
        opId,
        type: channel
          ? operationTypes.pendingMessage
          : operationTypes.pendingDirectMessage,
        meta: {
          message: message,
          channelId: channel ? channel.id : 'none',
          recipientAddress: channel ? 'none' : directChannelAddress,
          saveAdvert: !channel
        },
        checkConfirmationNumber: channel ? null : checkConfirmationNumber
      })
    )
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar(
        successNotification({ message: 'Offer successfully posted' })
      )
    )
  } catch (err) {
    console.log(err)
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar(
        errorNotification({
          message: "Couldn't send the message, please check node connection."
        })
      )
    )
  }
}

const handleSendTransfer = ({ values, history, payload }) => async (
  dispatch,
  getState
) => {
  const shippingData = identitySelectors.shippingData(getState())
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.ITEM_TRANSFER,
      data: {
        itemId: payload.id.substring(0, 64),
        tag: payload.tag,
        offerOwner: payload.offerOwner,
        shippingData: payload.provideShipping ? shippingData : null
      },
      spent: new BigNumber(values.zec)
    },
    privKey
  })
  dispatch(offersHandlers.epics.createOfferAdvert({ payload, history }))
  dispatch(
    directMessagesQueueHandlers.epics.addDirectMessage(
      {
        message,
        recipientAddress: payload.address,
        recipientUsername: payload.offerOwner
      },
      0,
      false
    )
  )
  dispatch(
    contactsHandlers.epics.updateDeletedChannelTimestamp({
      address: payload.id + payload.offerOwner,
      timestamp: 0
    })
  )
}

export const epics = {
  handleSend,
  handleSendTransfer
}

export default {
  epics
}
