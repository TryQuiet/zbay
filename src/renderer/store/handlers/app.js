import { ipcRenderer, remote } from 'electron'
import Immutable from 'immutable'
import { createAction, handleActions } from 'redux-actions'
import { actionTypes } from '../../../shared/static'
import { actionCreators } from './modals'

export const AppState = Immutable.Record(
  {
    version: null,
    transfers: Immutable.Map(),
    newUser: false,
    modalTabToOpen: null
  },
  'AppState'
)

export const initialState = AppState()

const loadVersion = createAction(actionTypes.SET_APP_VERSION, () =>
  remote.app.getVersion()
)
const setTransfers = createAction(actionTypes.SET_TRANSFERS)
const setModalTab = createAction(actionTypes.SET_CURRENT_MODAL_TAB)
const clearModalTab = createAction(actionTypes.CLEAR_CURRENT_MODAL_TAB)

export const actions = {
  loadVersion,
  setTransfers,
  setModalTab,
  clearModalTab
}

export const askForBlockchainLocation = () => async (dispatch, getState) => {
  dispatch(actionCreators.openModal('blockchainLocation')())
}

export const proceedWithSyncing = (payload) => async (dispatch, getState) => {
  ipcRenderer.send('proceed-with-syncing', payload)
  dispatch(actionCreators.closeModal('blockchainLocation')())
}

export const reducer = handleActions(
  {
    [loadVersion]: (state, { payload: version }) =>
      state.set('version', version),
    [setModalTab]: (state, { payload: tabName }) =>
      state.set('modalTabToOpen', tabName),
    [clearModalTab]: state => state.set('modalTabToOpen', null),
    [setTransfers]: (state, { payload: { id, value } }) => {
      return state.setIn(['transfers', id], value)
    }
  },
  initialState
)

export const epics = {
  askForBlockchainLocation,
  proceedWithSyncing
}

export default {
  epics,
  actions,
  reducer
}
