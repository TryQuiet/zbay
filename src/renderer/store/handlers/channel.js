import Immutable from 'immutable'
import BigNumber from 'bignumber.js'
import { createAction, handleActions } from 'redux-actions'

import history from '../../../shared/history'
import operationsHandlers, {
  operationTypes,
  PendingMessageOp
} from './operations'
import notificationsHandlers from './notifications'
import messagesQueueHandlers from './messagesQueue'
import messagesQueue from '../selectors/messagesQueue'
import messagesHandlers from './messages'
import channelsHandlers from './channels'
import offersHandlers from './offers'
import channelSelectors from '../selectors/channel'
import channelsSelectors from '../selectors/channels'
import identitySelectors from '../selectors/identity'
import logsHandlers from '../handlers/logs'
import { getClient } from '../../zcash'
import { messages } from '../../zbay'
import { errorNotification, LoaderState } from './utils'
import { channelToUri } from '../../zbay/channels'
import nodeSelectors from '../selectors/node'
import { getVault } from '../../vault'
import { messageType, actionTypes } from '../../../shared/static'

export const ChannelState = Immutable.Record(
  {
    spentFilterValue: new BigNumber(0),
    id: null,
    message: '',
    shareableUri: '',
    address: '',
    loader: LoaderState({ loading: false }),
    members: null,
    showInfoMsg: true
  },
  'ChannelState'
)

export const initialState = ChannelState()

const setSpentFilterValue = createAction(
  actionTypes.SET_SPENT_FILTER_VALUE,
  (_, value) => value
)
const setMessage = createAction(actionTypes.SET_CHANNEL_MESSAGE)
const setChannelId = createAction(actionTypes.SET_CHANNEL_ID)
const setLoading = createAction(actionTypes.SET_CHANNEL_LOADING)
const setLoadingMessage = createAction(actionTypes.SET_CHANNEL_LOADING_MESSAGE)
const setShareableUri = createAction(actionTypes.SET_CHANNEL_SHAREABLE_URI)
const setAddress = createAction(actionTypes.SET_CHANNEL_ADDRESS)
const resetChannel = createAction(actionTypes.SET_CHANNEL)

export const actions = {
  setLoading,
  setLoadingMessage,
  setSpentFilterValue,
  setMessage,
  setShareableUri,
  setChannelId,
  resetChannel
}

const loadChannel = id => async (dispatch, getState) => {
  try {
    dispatch(setChannelId(id))

    // Calculate URI on load, that way it won't be outdated, even if someone decides
    // to update channel in vault manually
    const channel = channelSelectors.data(getState()).toJS()
    const uri = await channelToUri(channel)
    dispatch(setShareableUri(uri))
    dispatch(setAddress(channel.address))
    await dispatch(clearNewMessages())
    await dispatch(updateLastSeen())
  } catch (err) {}
}
const loadOffer = (id, address) => async (dispatch, getState) => {
  try {
    await dispatch(offersHandlers.epics.updateLastSeen({ itemId: id }))
    dispatch(setChannelId(id))
    dispatch(setShareableUri(''))
    dispatch(setAddress(address))
  } catch (err) {}
}
const linkChannelRedirect = targetChannel => async (dispatch, getState) => {
  let channels = channelsSelectors.channels(getState())
  let channel = channels.data.find(
    channel => channel.get('address') === targetChannel.address
  )
  const identityId = identitySelectors.id(getState())
  const lastblock = nodeSelectors.latestBlock(getState())
  const fetchTreshold = lastblock - 2000
  if (channel) {
    history.push(`/main/channel/${channel.get('id')}`)
    return
  }
  try {
    try {
      await getVault().channels.importChannel(identityId, targetChannel)
      getClient().keys.importIVK({
        ivk: targetChannel.keys.ivk,
        rescan: 'yes',
        startHeight: fetchTreshold
      })
    } catch (error) {}
    dispatch(logsHandlers.epics.saveLogs({ type: 'APPLICATION_LOGS', payload: `Importing channel ${targetChannel}` }))
    await dispatch(channelsHandlers.actions.loadChannels(identityId))
    channels = channelsSelectors.channels(getState())
    channel = channels.data.find(
      channel => channel.get('address') === targetChannel.address
    )
    await dispatch(messagesHandlers.epics.fetchMessages(channel))
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar({
        message: `Successfully imported channel ${targetChannel.name}`,
        options: {
          variant: 'success'
        }
      })
    )
    history.push(`/main/channel/${channel.get('id')}`)
  } catch (err) {
    console.log(err)
  }
}
const sendOnEnter = (event, resetTab) => async (dispatch, getState) => {
  if (resetTab) {
    resetTab(0)
  }
  const enterPressed = event.nativeEvent.keyCode === 13
  const shiftPressed = event.nativeEvent.shiftKey === true
  const channel = channelSelectors.data(getState()).toJS()
  const messageToSend = channelSelectors.message(getState())
  const currentMessage = messagesQueue
    .queue(getState())
    .find(dm => dm.get('channelId') === channel.id)
  if (enterPressed && !shiftPressed) {
    event.preventDefault()
    const privKey = identitySelectors.signerPrivKey(getState())
    let message
    if (currentMessage !== undefined) {
      message = messages.createMessage({
        messageData: {
          type: messageType.BASIC,
          data:
            currentMessage.get('message').get('message') + '\n' + messageToSend
        },
        privKey: privKey
      })
    } else {
      message = messages.createMessage({
        messageData: {
          type: messageType.BASIC,
          data: messageToSend
        },
        privKey: privKey
      })
    }
    dispatch(
      messagesQueueHandlers.epics.addMessage({ message, channelId: channel.id })
    )
    dispatch(setMessage(''))
  }
}
const sendChannelSettingsMessage = ({
  address,
  minFee = '0',
  onlyRegistered = '0'
}) => async (dispatch, getState) => {
  const identityAddress = identitySelectors.address(getState())
  const owner = identitySelectors.signerPubKey(getState())
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.CHANNEL_SETTINGS,
      data: {
        owner,
        minFee,
        onlyRegistered
      }
    },
    privKey: privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: address,
    identityAddress
  })
  try {
    await getClient().payment.send(transfer)
  } catch (err) {
    notificationsHandlers.actions.enqueueSnackbar(
      dispatch(
        errorNotification({
          message: "Couldn't create channel, please check node connection."
        })
      )
    )
  }
}

const resendMessage = messageData => async (dispatch, getState) => {
  dispatch(operationsHandlers.actions.removeOperation(messageData.id))
  const identityAddress = identitySelectors.address(getState())
  const channel = channelSelectors.data(getState()).toJS()
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageData.type,
      data: messageData.message,
      spent: '0'
    },
    privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: channel.address,
    identityAddress
  })
  try {
    const opId = await getClient().payment.send(transfer)
    await dispatch(
      operationsHandlers.epics.observeOperation({
        opId,
        type: operationTypes.pendingMessage,
        meta: PendingMessageOp({
          channelId: channel.id,
          message: Immutable.fromJS(message)
        })
      })
    )
  } catch (err) {
    notificationsHandlers.actions.enqueueSnackbar(
      errorNotification({
        message: "Couldn't send the message, please check node connection."
      })
    )
  }
}

const updateLastSeen = () => async (dispatch, getState) => {
  const channelId = channelSelectors.channelId(getState())
  return dispatch(channelsHandlers.epics.updateLastSeen({ channelId }))
}

const clearNewMessages = () => async (dispatch, getState) => {
  const channelId = channelSelectors.channelId(getState())
  dispatch(messagesHandlers.actions.cleanNewMessages({ channelId }))
}

export const epics = {
  sendOnEnter,
  loadChannel,
  resendMessage,
  clearNewMessages,
  updateLastSeen,
  loadOffer,
  sendChannelSettingsMessage,
  linkChannelRedirect
}

// TODO: we should have a global loader map
export const reducer = handleActions(
  {
    [setLoading]: (state, { payload: loading }) =>
      state.setIn(['loader', 'loading'], loading),
    [setLoadingMessage]: (state, { payload: message }) =>
      state.setIn(['loader', 'message'], message),
    [setSpentFilterValue]: (state, { payload: value }) =>
      state.set('spentFilterValue', new BigNumber(value)),
    [setMessage]: (state, { payload: value }) => state.set('message', value),
    [setChannelId]: (state, { payload: id }) => state.set('id', id),
    [setShareableUri]: (state, { payload: uri }) =>
      state.set('shareableUri', uri),
    [setAddress]: (state, { payload: address }) =>
      state.set('address', address),
    [resetChannel]: () => initialState
  },
  initialState
)

export default {
  reducer,
  epics,
  actions
}
