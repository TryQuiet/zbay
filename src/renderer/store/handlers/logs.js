import Immutable from 'immutable'
import { createAction, handleActions } from 'redux-actions'
import { ipcRenderer } from 'electron'
import { actionTypes } from '../../../shared/static'

export const Logs = Immutable.Record(
  {
    transactionLogs: [],
    applicationLogs: [],
    nodeLogs: [],
    islogsFileLoaded: false,
    isLogWindowOpened: false
  },
  'logs'
)

export const initialState = Logs()

const setTransactionLogs = createAction(actionTypes.SET_TRANSACTIONS_LOGS)
const setApplicationLogs = createAction(actionTypes.SET_APPLICATIONS_LOGS)
const setNodeLogs = createAction(actionTypes.SET_NODE_LOGS)
const setLogWindowOpened = createAction(actionTypes.SET_LOG_WINDOW_OPENED)

const loadTargetLogs = () => async (dispatch, getState) => {
  ipcRenderer.send('load-logs')
  dispatch(setLogWindowOpened(true))
}

const closeLogsWindow = () => async (dispatch, getState) => {
  ipcRenderer.send('disable-load-logs')
  dispatch(setLogWindowOpened(false))
}

const saveLogs = (data) => async (dispatch, getState) => {
  ipcRenderer.send('save-to-log-file', data)
}

export const actions = {
  setTransactionLogs,
  setApplicationLogs,
  setNodeLogs,
  setLogWindowOpened
}

export const epics = {
  loadTargetLogs,
  closeLogsWindow,
  saveLogs
}

export const reducer = handleActions(
  {
    [setTransactionLogs]: (state, { payload: transactionLogs }) => state.setIn(['transactionLogs'], transactionLogs),
    [setApplicationLogs]: (state, { payload: ApplicationLogs }) => state.setIn(['applicationLogs'], ApplicationLogs),
    [setNodeLogs]: (state, { payload: setNodeLogs }) => state.setIn(['nodeLogs'], setNodeLogs),
    [setLogWindowOpened]: (state, { payload: logPanelStatus }) => state.setIn(['isLogWindowOpened'], logPanelStatus)
  },
  initialState
)

export default {
  actions,
  epics,
  reducer
}
