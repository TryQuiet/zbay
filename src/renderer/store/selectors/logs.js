import { createSelector } from 'reselect'

const store = s => s

export const logs = createSelector(store, state => state.get('logsData'))

const applicationLogs = createSelector(logs, a => a.applicationLogs)
const transactionsLogs = createSelector(logs, a => a.transactionLogs)
const nodeLogs = createSelector(logs, a => a.nodeLogs)
const islogsFileLoaded = createSelector(logs, a => a.islogsFileLoaded)
const isLogWindowOpened = createSelector(logs, a => a.isLogWindowOpened)

export default {
  applicationLogs,
  transactionsLogs,
  nodeLogs,
  islogsFileLoaded,
  isLogWindowOpened
}
