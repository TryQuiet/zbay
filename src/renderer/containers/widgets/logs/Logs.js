import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import logsHandlers from '../../../store/handlers/logs'
import logsSelectors from '../../../store/selectors/logs'

import Logs from '../../../components/widgets/logs/Logs'

export const mapStateToProps = state => ({
  debugLogs: logsSelectors.nodeLogs(state),
  applicationLogs: logsSelectors.applicationLogs(state),
  transactionsLogs: logsSelectors.transactionsLogs(state)
})

export const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      closeLogsWindow: logsHandlers.epics.closeLogsWindow
    },
    dispatch
  )

export const LogsContainer = ({ debugLogs, closeLogsWindow, applicationLogs, transactionsLogs, height }) => <Logs debugLogs={debugLogs} height={height} closeLogsWindow={closeLogsWindow} applicationLogs={applicationLogs} transactionsLogs={transactionsLogs} />

export default R.compose(
  connect(mapStateToProps, mapDispatchToProps)
)(LogsContainer)
