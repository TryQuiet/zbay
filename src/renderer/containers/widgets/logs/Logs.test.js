import Immutable from 'immutable'

import create from '../../../store/create'
import { mapDispatchToProps, mapStateToProps } from './Logs'
import { Logs } from '../../../store/handlers/logs'

describe('Logs ', () => {
  let store = null
  beforeEach(() => {
    store = create({
      initialState: Immutable.Map({
        logsData: Logs({
          data: Immutable.fromJS({
            transactionLogs: [],
            rpcCallsLogs: [],
            nodeLogs: [],
            islogsFileLoaded: false,
            isLogWindowOpened: false
          })
        })
      })
    })
    jest.clearAllMocks()
  })

  it('will receive right props', () => {
    const state = mapStateToProps(store.getState())
    expect(state).toMatchSnapshot()
  })

  it('will receive right actions', () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })
})
