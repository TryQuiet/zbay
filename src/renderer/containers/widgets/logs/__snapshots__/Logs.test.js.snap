// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs  will receive right actions 1`] = `
Object {
  "closeLogsWindow": [Function],
}
`;

exports[`Logs  will receive right props 1`] = `
Object {
  "applicationLogs": Array [],
  "debugLogs": Array [],
  "transactionsLogs": Array [],
}
`;
